synapse/handlers/room.py

```diff
@@ -1461,7 +1462,7 @@ async def get_event_for_timestamp(
                         ex.args,
                     )
 
-        if not local_event_id:
+        if not local_event_id or not local_event:
```

Comment:
I think it's impossible to have a `local_event_id` but no `local_event`?

```suggestion
        if not local_event_id:
```

Reply:
That's unfortunate.
We could add an `assert local_event is not None` below?

Reply:
> We could add an `assert local_event is not None` below?

Feels like that's not better though.

I'd rather throw that descriptive error if `local_event` is ever `None` and have the single logic branch.

Reply:
Added 👍

https://github.com/matrix-org/synapse/blob/f845f07123e1fe09449fabad467ff4c3159dbaf9/synapse/handlers/room.py#L1465-L1468

Reply:
Thanks, looks good!